### PR TITLE
[Merged by Bors] - refactor: removing the old homology API

### PR DESCRIPTION
**Pull request by joelriou**
The now redundant old homology API is removed. (It was already mostly unused.)

The only slightly non-trivial changes are in `CategoryTheory.Abelian.Exact` where some definitions and proofs have been refactored. The proof that the category of abelian groups satisfies Grothendieck's AB5 axiom had to be moved to a separate file `Algebra.Category.Grp.AB5`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Category/Grp/AB5.lean [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2023 David Kurniadi Angdinata. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata, Moritz Firsching, Nikolas Kuhn, Amelia Livingston
+-/
```
Comment (erdOne) [inline Mathlib/Algebra/Category/Grp/AB5.lean:5]: Is this header correct?

Comment (joelriou) [reply Mathlib/Algebra/Category/Grp/AB5.lean:5]: This is based on https://github.com/leanprover-community/mathlib4/pull/5597

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Can [this](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Homology/ShortComplex/Basic.html) TODO be removed now?

bors d+

**Comment by joelriou** [comment]
Retrying...

bors merge

**Pull request closed without merging**